client/extensions/woocommerce/state/data-layer/product-categories/index.js

```diff
@@ -83,6 +85,42 @@ export function handleProductCategoryUpdate( store, action ) {
 	);
 }
 
+export function handleProductCategoryDelete( store, action ) {
+	const { siteId, category, successAction, failureAction } = action;
+
+	// Filter out any id we might have.
+	const { id, ...categoryData } = category;
+
+	if ( 'number' !== typeof id ) {
+		store.dispatch(
+			setError( siteId, action, {
+				message: 'Attempting to delete product category which has a placeholder ID.',
+				category,
+			} )
+		);
+		return;
+	}
+
+	store.dispatch(
+		del(
+			siteId,
+			`products/categories/${ id }?force=true`,
```

Comment:
Curious - what happens without `force=true`?

Reply:
API responds with a error message. It's a weird WC API thing, that requires it to be set, even if an endpoint doesn't support trashing:

https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-product-categories